### PR TITLE
sending claim file directly to the collector

### PR DESCRIPTION
**Pull request by shirmoran**

### CODE REVIEW

File: cnf-certification-test/tnf_config.yml [hunk 1]
```diff
@@ -49,3 +49,5 @@ servicesignorelist:
   - "hazelcast-platform-controller-manager-service"
   - "hazelcast-platform-webhook-service"
   - "new-pro-controller-manager-metrics-service"
+createdBy: "default-created-by"
```
Comment (aabughosh) [inline cnf-certification-test/tnf_config.yml:52]: can u please give a more clear name 
what is the meeining of created by?

File: pkg/collector/collector.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	req, err := http.NewRequest("POST", endPoint, &buffer)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", w.FormDataContentType())
+
+	return req, nil
+}
+
+func SendClaimFileToCollector(claimFilePath, createdBy, partnerName string) error {
```
Comment (edcdavid) [inline pkg/collector/collector.go:57]: why not add the endpoint to this function's arguments and get the endpoint as an optional env variable

File: cnf-certification-test/tnf_config.yml (hunk 1 above)
Comment (sebrandon1) [reply cnf-certification-test/tnf_config.yml:52]: I agree I think Created By and Partner Name might be the same thing/idea?

Comment (shirmoran) [reply cnf-certification-test/tnf_config.yml:52]: Created By means who created the file - CI, QE, partner etc.
Partner Name specifies the current partner

Comment (shirmoran) [reply cnf-certification-test/tnf_config.yml:52]: maybe TestsRanBy will be more meaningful than CreatedBy?

File: pkg/collector/collector.go (hunk 2 above)
Comment (shirmoran) [reply pkg/collector/collector.go:57]: Do you mean setting endpoint as an env variable back in suite_test?

File: cnf-certification-test/tnf_config.yml (hunk 1 above)
Comment (bnshr) [reply cnf-certification-test/tnf_config.yml:52]: executedBy

File: pkg/provider/provider.go [hunk 3]
```diff
@@ -117,6 +117,8 @@ type TestEnvironment struct { // rename this with testTarget
 	DaemonsetFailedToSpawn bool
 	ScaleCrUnderTest       []ScaleObject
 	StorageClassList       []storagev1.StorageClass
+	CreatedBy              string
+	PartnerName            string
 }
 
 type MachineConfig struct {
```
Comment (bnshr) [inline pkg/provider/provider.go:124]: What is machine here?

Comment (shirmoran) [reply pkg/provider/provider.go:124]: Its not my addition

Comment (shirmoran) [reply pkg/provider/provider.go:124]: Or am I misunderstanding you question

Comment (bnshr) [reply pkg/provider/provider.go:124]: Ah, my bad. I thought we are in collector repo. All good.

File: cnf-certification-test/tnf_config.yml (hunk 1 above)
Comment (bnshr) [reply cnf-certification-test/tnf_config.yml:52]: if it takes little effort, change it.

File: pkg/configuration/configuration.go [hunk 4]
```diff
@@ -145,6 +145,10 @@ type TestConfiguration struct {
 	ValidProtocolNames                          []string `yaml:"validProtocolNames" json:"validProtocolNames"`
 	ServicesIgnoreList                          []string `yaml:"servicesignorelist" json:"servicesignorelist"`
 	DebugDaemonSetNamespace                     string   `yaml:"debugDaemonSetNamespace" json:"debugDaemonSetNamespace"`
+	// Collector's parameters
+	CollectorsEndPoint string `yaml:"collectorsEndPoint" json:"collectorsEndPoint"`
```
Comment (bnshr) [inline pkg/configuration/configuration.go:149]: CollectorAppEndPoint

**Review by bnshr — COMMENTED** [review]
a small change is requested, otherwise looks good.

**Review by edcdavid — COMMENTED** [review]
probably need to rebase this PR, there are a lot of non-related changes

**Review by bnshr — APPROVED** [review]

File: cnf-certification-test/suite_test.go [hunk 5]
```diff
@@ -203,6 +204,12 @@ func TestTest(t *testing.T) {
 	claimOutputFile := filepath.Join(*claimPath, results.ClaimFileName)
 	claimhelper.WriteClaimOutput(claimOutputFile, payload)
 
+	// Send claim file to the collector
```
Comment (sebrandon1) [inline cnf-certification-test/suite_test.go:207]: Can this be put behind a config option?  Similar to the section below:

`!configuration.GetTestParameters().OmitArtifactsZipFile`

**Review by sebrandon1 — APPROVED** [review]
Approved!  Let's discuss how we can test enabling this feature in the CI to test the delivery of the claim json's.

**Review by aabughosh — APPROVED** [review]

**Pull request merged**